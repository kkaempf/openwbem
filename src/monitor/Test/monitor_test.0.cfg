unpriv_user { owprovdr }

read_dir {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/
  /dir/doesnt/exist/
}

read_link {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-exists
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-doesnt-exist
}

rename_from {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2
  /dir/doesnt/exist/rename
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file
}

rename_to {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
  /dir/doesnt/exist/rename-to
}

unlink {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
  /dir/doesnt/exist/unlink
}

open_r {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
}

open_rw {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-*
}

open_w {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-*
}

open_a {
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-*
}
