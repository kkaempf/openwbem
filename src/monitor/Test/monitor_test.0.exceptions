PrivilegeManagerException  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/
PrivilegeManagerException  FileSystemException: /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-doesnt-exist
PrivilegeManagerException  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 to /dir/doesnt/exist/rename-to
PrivilegeManagerException  CheckException: PrivilegeManager::rename: could not rename /dir/doesnt/exist/rename to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
PrivilegeManagerException  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
PrivilegeManagerException  CheckException: PrivilegeManager::open: "/montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-nonesuch"
FileSystemException  CheckException: PrivilegeManager::lstat: "/dir/doesnt/exist"
FileSystemException  CheckException: PrivilegeManager::stat: "/dir/doesnt/exist"
FileSystemException  CheckException: PrivilegeManager::stat: "/montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/dangling"
FileSystemException  CheckException: PrivilegeManager::lstat: "/montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/stat-doesnt-exist"
FileSystemException  CheckException: PrivilegeManager::stat: "/montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/stat-doesnt-exist"
