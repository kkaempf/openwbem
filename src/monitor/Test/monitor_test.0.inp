# These test cases where there are no protocol errors

readDirectory /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ keep_special
readDirectory /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/ omit_special
readDirectory /dir/doesnt/exist/ keep_special
readDirectory /dir/doesnt/exist/ omit_special

readLink /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-exists
readLink /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-doesnt-exist

rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename1 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 /dir/doesnt/exist/rename-to
rename /dir/doesnt/exist/rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to
rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

unlink /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-exists
unlink /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/unlink-doesnt-exist
unlink /dir/doesnt/exist/unlink

open in 0 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
open out 604 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-exists
  piffle-prog
open out 606 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-create
  hack-phlegm
open out|trunc 604
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-exists
  enigmatic
open out|trunc 604
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/out-trunc-create
  wizard-of-oz
open out|app 700
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-exists
  send-in-the-clowns
open out|app 500 
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/app-create
  horatio-at-the-bridge
open in|out 404 
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists
  xxx
open in|out 400 
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-nonesuch
  foo-bar-baz
open in|out|trunc 606 
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-exists
  yyy
open in|out|trunc 606 
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-trunc-create
  This-is-the-song-that-never-ends
open in|out|ate 402
  /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/in-out-exists-ate
  xxx
open in|nonblock 0 /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/bar
