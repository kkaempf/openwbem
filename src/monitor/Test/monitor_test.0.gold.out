User: owprovdr

readDirectory
  .
  ..
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  link-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  these-are-the-link-contents
  unlink-exists

readDirectory
  app-exists
  bar
  foo
  in-out-exists
  in-out-exists-ate
  in-out-trunc-exists
  link-exists
  out-exists
  out-trunc-exists
  rename1
  rename2
  these-are-the-link-contents
  unlink-exists

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

readDirectory
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::readDirectory: could not read /dir/doesnt/exist/

readLink
  ./these-are-the-link-contents

readLink
Exception:
  type: PrivilegeManagerException
  msg:  FileSystemException: /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/link-doesnt-exist: 2(No such file or directory)

rename

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename2 to /dir/doesnt/exist/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /dir/doesnt/exist/rename to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

rename
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::rename: could not rename /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/no-such-file to /montest-689acb0e1ec89f45-7085a2a1780f5f42/test_dir/rename-to

unlink
1

unlink
0

unlink
0

open
  Hokey smokes, Bullwinkle!


open

open

open

open

open

open

open
  xxxtle Bo Peep has lost her sheep


open
Exception:
  type: PrivilegeManagerException
  msg:  CheckException: PrivilegeManager::open: 2(No such file or directory)

open
  yyy

open
  This-is-the-song-that-never-ends

open
  Little Bo Peep has lost her sheep
xxx
